### PR TITLE
fix: add-next-in-beta-plugin-meetings

### PR DESCRIPTION
**Pull request by Shreyas281299**
# COMPLETES # [SPARK-513490](https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-513490)

## This pull request addresses

This Pull request pulls all the plugin-meeting changes from the next branch . with  bug fixes, transcriptions etc. 
Makes sure there are no un necessary changes in the code allowing us to merger this to beta branch before web client moves to next branch after doing test on web client test env

## by making the following changes

PRs for Voicea
Base PR - feat(plugin-meetings): replace transcription with voicea plugin #3336
Voicea UT in meeting/index.ts - fix(plugin-meetings): meeting transcription UT revamp Shreyas281299/webex-js-sdk#12
Voicea Utils UT - fix(plugin-meetings): voicea utils test #3452
BEMS Bug fix - fix(meetings): fix to avoid llm connection update when its not enabled #3479
Fixed browser and unit test - https://github.com/webex/webex-js-sdk/pull/3456
https://github.com/webex/webex-js-sdk/pull/3378
https://github.com/meta-dipanshu-sharma/webex-js-sdk/pull/8
https://github.com/webex/webex-js-sdk/pull/2731

### Reviewed by web-client team
Voicea for Guest users & stopTranscription - fix(plugin-meetings): listen to transcript for guests #3405
https://github.com/webex/webex-js-sdk/pull/2953

### Change Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Tooling change
- [ ] Internal code refactor

## The following scenarios where tested

< ENUMERATE TESTS PERFORMED, WHETHER MANUAL OR AUTOMATED >

### I certified that

- [ ] I have read and followed [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request)
- [ ] I discussed changes with code owners prior to submitting this pull request

- [ ] I have not skipped any automated checks
- [ ] All existing and new tests passed
- [ ] I have updated the documentation accordingly

---

Make sure to have followed the [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request) before submitting.

### CODE REVIEW

**Comment by arun3528** [comment]
PR links needs 


plugin-meeting/Constant.js
JWT change
Voicia
Transcrioption
meetinginfo v2
interpretation
sendNamedMEdiaGroup in meeting/index

File: packages/@webex/plugin-meetings/package.json [hunk 1]
```diff
@@ -67,7 +67,7 @@
     "global": "^4.4.0",
     "ip-anonymize": "^0.1.0",
     "javascript-state-machine": "^3.1.0",
-    "jsonwebtoken": "^8.5.1",
+    "jwt-decode": "3.1.2",
```
Comment (arun3528) [inline packages/@webex/plugin-meetings/package.json:70]: https://github.com/webex/webex-js-sdk/pull/3378

File: packages/@webex/plugin-meetings/src/constants.ts [hunk 2]
```diff
@@ -48,6 +48,8 @@ export const MEETINGINFO = 'meetingInfo';
 export const MEET = 'meet';
 export const MEET_M = 'm';
 export const MEDIA = 'media';
+export const MEET_CO = 'co';
+export const MEET_CISCO = 'cisco';
```
Comment (arun3528) [inline packages/@webex/plugin-meetings/src/constants.ts:52]: https://github.com/meta-dipanshu-sharma/webex-js-sdk/pull/8

File: packages/@webex/plugin-meetings/src/meeting-info/utilv2.ts [hunk 3]
```diff
@@ -68,6 +68,8 @@ MeetingInfoUtil.isMeetingLink = (value: string) => {
     parsedUrl.pathname &&
     (parsedUrl.pathname.includes(`/${MEET}`) ||
       parsedUrl.pathname.includes(`/${MEET_M}`) ||
+      parsedUrl.pathname.includes(`/${MEET_CISCO}`) ||
```
Comment (arun3528) [inline packages/@webex/plugin-meetings/src/meeting-info/utilv2.ts:71]: https://github.com/meta-dipanshu-sharma/webex-js-sdk/pull/8

File: packages/@webex/plugin-meetings/src/meeting/index.ts [hunk 4]
```diff
@@ -1,6 +1,6 @@
 import uuid from 'uuid';
 import {cloneDeep, isEqual, isEmpty} from 'lodash';
-import jwt from 'jsonwebtoken';
+import jwtDecode from 'jwt-decode';
```
Comment (arun3528) [inline packages/@webex/plugin-meetings/src/meeting/index.ts:3]: https://github.com/webex/webex-js-sdk/pull/3378

File: packages/@webex/plugin-meetings/src/meeting/index.ts [hunk 5]
```diff
@@ -3616,7 +3738,7 @@ export default class Meeting extends StatelessWebexPlugin {
    * @returns {void}
    */
   public setPermissionTokenPayload(permissionToken: string) {
-    this.permissionTokenPayload = jwt.decode(permissionToken);
+    this.permissionTokenPayload = jwtDecode(permissionToken);
```
Comment (arun3528) [inline packages/@webex/plugin-meetings/src/meeting/index.ts:3741]: https://github.com/webex/webex-js-sdk/pull/3378

File: packages/@webex/plugin-meetings/src/meeting-info/meeting-info-v2.ts [hunk 6]
```diff
@@ -204,22 +204,16 @@ export default class MeetingInfoV2 {
...
-      .then((conversation) => {
-        const body: {
-          title: string;
-          spaceUrl: string;
-          keyUrl: string;
-          kroUrl: string;
-          invitees: any[];
-          installedOrgID?: string;
-        } = {
+    return this.webex
```
Comment (Shreyas281299) [inline packages/@webex/plugin-meetings/src/meeting-info/meeting-info-v2.ts:207]: https://github.com/webex/webex-js-sdk/pull/2731

File: packages/@webex/plugin-meetings/test/unit/spec/meeting/voicea-meeting.ts [hunk 7]
```diff
@@ -0,0 +1,266 @@
+import {
```
Comment (mkesavan13) [inline packages/@webex/plugin-meetings/test/unit/spec/meeting/voicea-meeting.ts:1]: * Base PR - https://github.com/webex/webex-js-sdk/pull/3336
* Voicea for Guest users & stopTranscription - https://github.com/webex/webex-js-sdk/pull/3405
* Voicea UT in meeting/index.ts - https://github.com/Shreyas281299/webex-js-sdk/pull/12
* Voicea Utils UT - https://github.com/webex/webex-js-sdk/pull/3452

File: packages/@webex/plugin-meetings/test/unit/spec/meeting/utils.js [hunk 8]
```diff
@@ -71,6 +72,37 @@ describe('plugin-meetings', () => {
...
+        assert.calledOnce(meeting.closePeerConnections);
+
+        assert.calledOnce(meeting.unsetRemoteStreams);
+        assert.calledOnce(meeting.unsetPeerConnections);
+        assert.calledOnce(meeting.reconnectionManager.cleanUp);
+        assert.calledOnce(meeting.stopKeepAlive);
+        assert.notCalled(meeting.updateLLMConnection);
+        assert.calledOnce(meeting.breakouts.cleanUp);
+        assert.calledOnce(meeting.simultaneousInterpretation.cleanUp);
+      });
```
Comment (mkesavan13) [inline packages/@webex/plugin-meetings/test/unit/spec/meeting/utils.js:105]: - BEMS Bug fix - https://github.com/webex/webex-js-sdk/pull/3479

File: packages/@webex/plugin-meetings/src/meeting-info/util.ts [hunk 9]
```diff
@@ -221,12 +220,16 @@ MeetingInfoUtil.generateOptions = async (from) => {
     try {
       await webex.internal.services.waitForCatalog('postauth');
 
-      const conversationUrl = webex.internal.conversation.getUrlFromClusterId({
-        cluster: hydraId.cluster,
-        id: hydraId.destination,
-      });
+      const serviceUrl = webex.internal.services.getServiceUrlFromClusterId(
```
Comment (Shreyas281299) [inline packages/@webex/plugin-meetings/src/meeting-info/util.ts:223]: https://github.com/webex/webex-js-sdk/pull/2731

File: packages/@webex/plugin-meetings/src/meeting-info/utilv2.ts [hunk 10]
```diff
@@ -162,7 +164,13 @@ MeetingInfoUtil.getDestinationType = async (from) => {
     };
   }
   const options: any = {};
-  const hydraId = MeetingInfoUtil.getHydraId(destination);
+  let hydraId;
```
Comment (Shreyas281299) [inline packages/@webex/plugin-meetings/src/meeting-info/utilv2.ts:167]: https://github.com/webex/webex-js-sdk/pull/2731

File: packages/@webex/plugin-meetings/test/unit/spec/meeting-info/meetinginfov2.js [hunk 11]
```diff
@@ -795,13 +795,17 @@ describe('plugin-meetings', () => {
       it('Make a request to /spaceInstant when conversationUrl', async () => {
         const {invitee} = setup();
 
-        const result = await meetingInfo.createAdhocSpaceMeeting(conversationUrl);
+        webex.request.resolves({
```
Comment (Shreyas281299) [inline packages/@webex/plugin-meetings/test/unit/spec/meeting-info/meetinginfov2.js:798]: https://github.com/webex/webex-js-sdk/pull/2731

File: packages/@webex/plugin-meetings/test/unit/spec/meetings/index.js [hunk 12]
```diff
@@ -1986,7 +2053,8 @@ describe('plugin-meetings', () => {
         webex.meetings.meetingInfo.fetchMeetingInfo = sinon.stub().returns(
           Promise.resolve({
             body: {
-              permissionToken: 'PT',
+              permissionToken:
```
Comment (Shreyas281299) [inline packages/@webex/plugin-meetings/test/unit/spec/meetings/index.js:2056]: https://github.com/webex/webex-js-sdk/pull/3456

File: packages/@webex/plugin-meetings/test/unit/spec/meetings/index.js [hunk 13]
```diff
@@ -1356,7 +1364,8 @@ describe('plugin-meetings', () => {
 
           it('accepts injected meeting info with meeting lookup url', async () => {
             const meetingInfo = {
-              permissionToken: 'PT',
+              permissionToken:
```
Comment (Shreyas281299) [inline packages/@webex/plugin-meetings/test/unit/spec/meetings/index.js:1367]: https://github.com/webex/webex-js-sdk/pull/3456

File: packages/@webex/plugin-meetings/test/unit/spec/meetings/index.js [hunk 14]
```diff
@@ -625,7 +626,11 @@ describe('plugin-meetings', () => {
                   loci: [{url: 'still-valid-locus-url'}],
                 })
               );
-              MeetingUtil.cleanUp = sinon.stub().returns(Promise.resolve());
+              cleanUpSpy = sinon.stub(MeetingUtil, 'cleanUp').returns(Promise.resolve());
```
Comment (Shreyas281299) [inline packages/@webex/plugin-meetings/test/unit/spec/meetings/index.js:629]: https://github.com/webex/webex-js-sdk/pull/3456

File: packages/@webex/plugin-meetings/test/unit/spec/meeting/utils.js [hunk 15]
```diff
@@ -369,7 +401,7 @@ describe('plugin-meetings', () => {
           getWebexObject: sinon.stub().returns(webex),
         };
 
-        MeetingUtil.parseLocusJoin = sinon.stub();
+        const parseLocusJoinSpy = sinon.stub(MeetingUtil, 'parseLocusJoin');
```
Comment (Shreyas281299) [inline packages/@webex/plugin-meetings/test/unit/spec/meeting/utils.js:404]: https://github.com/webex/webex-js-sdk/pull/3456

File: packages/@webex/plugin-meetings/test/unit/spec/media/index.ts [hunk 16]
```diff
@@ -7,6 +7,10 @@ import {forEach} from 'lodash';
 import MockWebex from '@webex/test-helper-mock-webex';
 
 describe('createMediaConnection', () => {
+  let clock;
```
Comment (Shreyas281299) [inline packages/@webex/plugin-meetings/test/unit/spec/media/index.ts:10]: https://github.com/webex/webex-js-sdk/pull/3456

File: packages/@webex/plugin-meetings/src/member/member.types.ts [hunk 17]
```diff
@@ -0,0 +1,13 @@
+// values are inherited from locus so don't update these
```
Comment (Shreyas281299) [inline packages/@webex/plugin-meetings/src/member/member.types.ts:1]: https://github.com/webex/webex-js-sdk/pull/2953

File: packages/@webex/plugin-meetings/src/meeting-info/meeting-info-v2.ts [hunk 18]
```diff
@@ -204,22 +204,16 @@ export default class MeetingInfoV2 {
...
+      .request({uri: conversationUrl, qs: {includeParticipants: true}, disableTransform: true})
+      .then(({body: conversation}) => {
+        const body = {
           title: conversation.displayName,
           spaceUrl: conversation.url,
           keyUrl: conversation.encryptionKeyUrl,
           kroUrl: conversation.kmsResourceObjectUrl,
           invitees: getInvitees(conversation.participants?.items),
+          installedOrgID,
```
Comment (marcin-bazyl) [inline packages/@webex/plugin-meetings/src/meeting-info/meeting-info-v2.ts:216]: adding `installedOrgID` here means that lines 219-221 can be removed, but it means a slight change in behavior, for example if installedOrgID was an empty string, before this PR it wouldn't be set in the body and now it will be. @Coread do you know if that's OK?

File: packages/@webex/plugin-meetings/src/meeting/util.ts [hunk 19]
```diff
@@ -202,7 +202,11 @@ const MeetingUtil = {
         meeting.reconnectionManager.cleanUp();
       })
       .then(() => meeting.stopKeepAlive())
-      .then(() => meeting.updateLLMConnection());
+      .then(() => {
+        if (meeting.config?.enableAutomaticLLM) {
+          meeting.updateLLMConnection();
```
Comment (marcin-bazyl) [inline packages/@webex/plugin-meetings/src/meeting/util.ts:207]: is the `return` here missing on purpose?

File: packages/@webex/plugin-meetings/src/member/member.types.ts [hunk 20]
```diff
@@ -0,0 +1,13 @@
...
+export enum MediaStatus {
```
Comment (marcin-bazyl) [inline packages/@webex/plugin-meetings/src/member/member.types.ts:2]: the same enum and also same IMediaStatus interface are already defined in packages/@webex/plugin-meetings/src/member/types.ts file

File: packages/@webex/plugin-meetings/test/integration/spec/journey.js [hunk 21]
```diff
@@ -871,7 +871,7 @@ skipInNode(describe)('plugin-meetings', () => {
           assert.equal(bob.meeting.shareStatus, 'whiteboard_share_active');
         }));
 
-      it('alice adds chris as guest to 1:1 meeting', () =>
+      it('alice adds chris as guest to 1:1 meeting', async() =>
```
Comment (marcin-bazyl) [inline packages/@webex/plugin-meetings/test/integration/spec/journey.js:874]: why is this needed if there is no `await` in this function and it already returns a promise anyway (the result of Promise.all() call)

File: packages/@webex/plugin-meetings/test/integration/spec/space-meeting.js [hunk 22]
```diff
@@ -34,7 +34,7 @@ skipInNode(describe)('plugin-meetings', () => {
   describe('space-meeting', () => {
     let space = null;
 
-    before(() =>
+    before(async () =>
```
Comment (marcin-bazyl) [inline packages/@webex/plugin-meetings/test/integration/spec/space-meeting.js:37]: same as above - why is this needed, how does it fix anything?

File: packages/@webex/plugin-meetings/src/meeting/util.ts (hunk 19 above)
Comment (sreenara) [reply packages/@webex/plugin-meetings/src/meeting/util.ts:207]: @Kesari3008 can you answer the question above?

File: packages/@webex/plugin-meetings/test/integration/spec/space-meeting.js (hunk 22 above)
Comment (Shreyas281299) [reply packages/@webex/plugin-meetings/test/integration/spec/space-meeting.js]: Removed from next, cherry-picked here.

Comment (Shreyas281299) [reply packages/@webex/plugin-meetings/test/integration/spec/space-meeting.js]: Yes @arun3528 . I raised a PR in next first then cherry-picked that here, there the build was green and locally also everything passed.. 
PR - https://github.com/webex/webex-js-sdk/pull/3501

File: packages/@webex/plugin-meetings/src/meeting/util.ts (hunk 19 above)
Comment (Kesari3008) [reply packages/@webex/plugin-meetings/src/meeting/util.ts:207]: @marcin-bazyl There has been no return statement present here from the start. This is the cleanup function where we needed to added this check to avoid updating LLM connection when it's not enabled.
You can refer to PR: https://github.com/webex/webex-js-sdk/pull/3479 where this change has gone in for more details.

File: packages/@webex/plugin-meetings/src/meeting-info/meeting-info-v2.ts (hunk 18 above)
Comment (Coread) [reply packages/@webex/plugin-meetings/src/meeting-info/meeting-info-v2.ts:216]: installedOrgID will be checked for policies. Presumably an orgID of blank will not match any policies and so will not do anything. Not sure the reason for this change though

File: packages/@webex/plugin-meetings/src/member/member.types.ts (hunk 20 above)
Comment (Shreyas281299) [reply packages/@webex/plugin-meetings/src/member/member.types.ts]: Removed this file.
PR in next for this changes - https://github.com/webex/webex-js-sdk/pull/3513

File: packages/@webex/plugin-meetings/src/meeting/util.ts (hunk 19 above)
Comment (Coread) [reply packages/@webex/plugin-meetings/src/meeting/util.ts:207]: This was the wrong approach. updateLLMConnection was designed to be idempotent, so you could call it from wherever and it wouldn't be a problem. enableAutomaticLLM was not the only time where LLM could be enabled. The intention was to be able to manually enable LLM later on in the meeting flow, in which case a cleanup would still be required. That being said, it isn't a large enough issue to want to fix in this PR.

**Review by Coread — APPROVED** [review]

**Review by sreenara — APPROVED** [review]

**Review by marcin-bazyl — APPROVED** [review]
